modules/navigation/2d/godot_navigation_server_2d.cpp

```diff
@@ -456,16 +452,48 @@ Vector<Vector2> GodotNavigationServer2D::obstacle_get_vertices(RID p_obstacle) c
 	return vector_v3_to_v2(NavigationServer3D::get_singleton()->obstacle_get_vertices(p_obstacle));
 }
 
-void GodotNavigationServer2D::query_path(const Ref<NavigationPathQueryParameters2D> &p_query_parameters, Ref<NavigationPathQueryResult2D> p_query_result) const {
+void GodotNavigationServer2D::query_path(const Ref<NavigationPathQueryParameters2D> &p_query_parameters, Ref<NavigationPathQueryResult2D> p_query_result, const Callable &p_callback) {
 	ERR_FAIL_COND(!p_query_parameters.is_valid());
 	ERR_FAIL_COND(!p_query_result.is_valid());
 
-	const NavigationUtilities::PathQueryResult _query_result = NavigationServer3D::get_singleton()->_query_path(p_query_parameters->get_parameters());
+	Ref<NavigationPathQueryParameters3D> query_parameters;
```

Comment:
All parameter-like objects exposed to users and scripting are RefCounted. For once because users cant be expected to manage and free those manually and also because the script binding system can not bind to more native C++ things.

That code here in particular is just a workaround because there is no real 2D NavigationServer so we need to create 3D objects behind the scene to query the NavigationServer3D.